applications/pelion_client/sample.yaml

```diff
@@ -0,0 +1,14 @@
+sample:
+  name: Pelion Client Application
+  description: Pelion conectivity reference application
+tests:
+  applications.pelion_client.zdebug:
+    build_only: true
+    extra_args: -DEXTRA_CFLAGS="-Werror -Wno-error=int-to-pointer-cast -Wno-implicit-function-declaration -Wno-error=format -Wno-error=pointer-to-int-cast -Wno-error=cpp -Wno-error=discarded-qualifiers -Wno-error=uninitialized -Wno-error=unused-function -Wno-error=deprecated-declarations"
+    platform_allow: nrf52840dk_nrf52840 nrf9160dk_nrf9160ns nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuappns
+    integration_platforms:
+      - nrf52840dk_nrf52840
+      - nrf9160dk_nrf9160ns
```

Comment:
Sorry, just seen it, the names were changed in the upstream, and in NCS with the recent upmerge. nrf9160dk_nrf9160ns -> nrf9160dk_nrf9160_ns and nrf5340dk_nrf5340_cpuappns -> nrf5340dk_nrf5340_cpuapp_ns 
It seems there is also still something wrong. The app wasn't build
```
15:59:11  ERROR   - Cmake build failure: /home/work/nrf/applications/pelion_client for nrf5340dk_nrf5340_cpuapp
15:59:11  Process Process-5:
15:59:11  Traceback (most recent call last):
15:59:11    File "/usr/local/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
15:59:11      self.run()
15:59:11    File "/usr/local/lib/python3.8/multiprocessing/process.py", line 108, in run
15:59:11      self._target(*self._args, **self._kwargs)
15:59:11    File "/home/work/zephyr/scripts/pylib/twister/twisterlib.py", line 3350, in pipeline_mgr
15:59:11      pb.process(pipeline, done_queue, task, lock, results)
15:59:11    File "/home/work/zephyr/scripts/pylib/twister/twisterlib.py", line 2302, in process
15:59:11      res = self.cmake()
15:59:11    File "/home/work/zephyr/scripts/pylib/twister/twisterlib.py", line 2541, in cmake
15:59:11      res = self.run_cmake(args)
15:59:11    File "/home/work/zephyr/scripts/pylib/twister/twisterlib.py", line 2067, in run_cmake
15:59:11      with open(os.path.join(self.build_dir, self.log), "a") as log:
15:59:11  FileNotFoundError: [Errno 2] No such file or directory: '/home/work/twister-out/nrf5340dk_nrf5340_cpuapp/pelion_client/applications.pelion_client.zdebug/build.log'
```
It seems that the `extra_args` you used are wrongly parsed and cmake fails. I will come back with the proper syntax for twister.
I will investigate it also from the twister side, why this was not counted as an error in the CI.

Reply:
this is working when passed to west
so I would expect if CI just adds it to the build line (as it does) for build types, it should work